Automattic/wp-calypso
# Framework: Refactor sites Redux state to resolve ambiguity and reflect recommendations

Related: #2080

This pull request seeks to resolve a few inconsistencies with the hope of providing more clarity in the usage of sites Redux state. Specifically, this targets:

- Reflecting the UI state as tracking the selected site's ID by renaming `selectedSite` to `selectedSiteId`
- Renaming the site selection action type from `SET_SELECTED_SITE` to `SELECTED_SITE_SET` per [recent guideline decisions](https://github.com/Automattic/wp-calypso/commit/aff948ec88f67dd58ddc052db75ac5199355488b) ([related discussion](https://github.com/Automattic/wp-calypso/pull/1660/files#r47770586))

__Testing instructions:__

This impacts setting the selected site in the My Sites controller. To my knowledge, the currently selected site object is only requested in the post editor sharing accordion, so confirm that the testing instructions from #756 still succeed.

Also confirm that Mocha tests pass by running `make test` from the project root directory or directly from `client/state/ui` and `client/state/sites`.

/cc @gwwar
